src/ServicePulse.Host/vue/src/components/failedmessages/FlowDiagram.vue

```diff
@@ -90,12 +112,12 @@ function constructNodes(mappedMessages) {
         return group.reduce(
           ({ result, currentWidth, previousParent }, message) => {
             //position on current level needs to be based on parent Node, so see if one exists
-            const parentMessage = previousLevel?.find((plMessage) => message.parentId === plMessage.messageId && message.parentEndpoint === plMessage.receivingEndpoint);
+            const parentMessage = previousLevel?.find((plMessage) => message.parentId === plMessage.messageId && message.parentEndpoint === plMessage.receivingEndpoint) ?? null;
             //if the current parent node is the same as the previous parent node, then the current position needs to be to the right of siblings
             const currentParentWidth = previousParent === parentMessage ? currentWidth : 0;
-            const startX = parentMessage == null ? 0 : parentMessage.XPos - parentMessage.width / 2;
+            const startX = parentMessage == null ? 0 : (parentMessage.XPos as number) - (parentMessage.width as number) / 2;
```

Comment:
I see now what you were talking about @PhilBastian.
In this case use the assertion operator, https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator